staging/src/k8s.io/apiserver/pkg/server/signal.go

```diff
@@ -67,3 +70,35 @@ func RequestShutdown() bool {
 
 	return false
 }
+
+func dumpStacks(writeToFile bool, path string) {
+	var (
+		buf       []byte
+		stackSize int
+	)
+	bufferLen := 16384
+	for stackSize == len(buf) {
+		buf = make([]byte, bufferLen)
+		stackSize = runtime.Stack(buf, true)
+		bufferLen *= 2
+	}
+	buf = buf[:stackSize]
+	if writeToFile {
```

Comment:
Why we need an condition `writeToFile` here? If we don't need to write to file, so why dumps goroutine then. 🤔